src/Adapter/ContainerFinder.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter;
+
+use Context;
+use Controller;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Find the container
+ */
+class ContainerFinder
+{
+    /**
+     * @param Context $context
+     *
+     * @return ContainerBuilder|ContainerInterface|null
+     */
+    public function getContainer(Context $context)
+    {
+        $container = isset($context->container) ? $context->container : null;
+        if (null !== $container) {
+            return $container;
+        }
+        $container = isset($context->controller) && $context->controller instanceof Controller ? $context->controller->getContainer() : null;
+        if (null !== $container) {
+            return $container;
+        }
+
+        return SymfonyContainer::getInstance();
```

Comment:
I think it would be a good idea to throw an exception if this is null

Reply:
Since @quetzacoalt91 pr, Symfony Container::getInstance() can't be null

Reply:
@eternoendless Isn't it unsafe to replace a behavior where there are no exceptions to a case where there is ?

Reply:
It can if the kernel is not available and if the request does not go though a controller :trollface:

Reply:
> Isn't it unsafe to replace a behavior where there are no exceptions to a case where there is ?

In general, I believe that class should do what it's asked to, and throw an Exception if it's unable to do it. This technique is also beneficial for identifying unexpected errors that could go by unnoticed otherwise.

Reply:
:ok_hand: